modules/sdp/LocalSdpMunger.js

```diff
@@ -28,77 +25,75 @@ export default class LocalSdpMunger {
     constructor(tpc, localEndpointId) {
         this.tpc = tpc;
         this.localEndpointId = localEndpointId;
-        this.audioSourcesToMsidMap = new Map();
-        this.videoSourcesToMsidMap = new Map();
-    }
-
-    /**
-     * Returns a string that can be set as the MSID attribute for a source.
-     *
-     * @param {string} mediaType - Media type of the source.
-     * @param {string} trackId - Id of the MediaStreamTrack associated with the source.
-     * @param {string} streamId - Id of the MediaStream associated with the source.
-     * @returns {string|null}
-     */
-    _generateMsidAttribute(mediaType, trackId, streamId) {
-        if (!(mediaType && trackId)) {
-            logger.error(`Unable to munge local MSID - track id=${trackId} or media type=${mediaType} is missing`);
-
-            return null;
-        }
-        const pcId = this.tpc.id;
-
-        return `${streamId}-${pcId} ${trackId}-${pcId}`;
     }
 
     /**
-     * Updates or adds a 'msid' attribute in the format '<endpoint_id>-<mediaType>-<trackIndex>-<tpcId>'
-     * example - d8ff91-video-0-1
-     * All other attributes like 'cname', 'label' and 'mslabel' are removed since these are not processed by Jicofo.
+     * Updates or adds a 'msid' attribute in the format. All other attributes like 'cname', 'label' and 'mslabel' are
+     * removed since these are not processed by Jicofo.
      *
      * @param {MLineWrap} mediaSection - The media part (audio or video) of the session description which will be
      * modified in place.
      * @returns {void}
      * @private
      */
-    _transformMediaIdentifiers(mediaSection) {
-        const mediaType = mediaSection.mLine?.type;
-        const mediaDirection = mediaSection.mLine?.direction;
-        const msidLine = mediaSection.mLine?.msid;
-        const sources = [ ...new Set(mediaSection.mLine?.ssrcs?.map(s => s.id)) ];
-        const streamId = `${this.localEndpointId}-${mediaType}`;
-        let trackId = msidLine ? msidLine.split(' ')[1] : `${this.localEndpointId}-${mediaSection.mLine.mid}`;
-
-        // Always overwrite msid since we want the msid to be in this format even if the browser generates one.
-        for (const source of sources) {
-            const msid = mediaSection.ssrcs.find(ssrc => ssrc.id === source && ssrc.attribute === 'msid');
-
-            if (msid) {
-                trackId = msid.value.split(' ')[1];
+    _transformMediaIdentifiers(mediaSection, ssrcMap) {
+        const mediaType = mediaSection.mLine.type;
+        const mediaDirection = mediaSection.mLine.direction;
+        const sources = [ ...new Set(mediaSection.mLine.ssrcs?.map(s => s.id)) ];
+        let sourceName;
+
+        if (ssrcMap.size) {
+            for (const [ id, trackSsrcs ] of ssrcMap.entries()) {
+                if (isEqual([ ...sources ].sort(), [ ...trackSsrcs.ssrcs ].sort())) {
```

Comment:
You are sorting a copy of the `sources` array on each loop iteration... that's 1 too-many copies. You can just sort the array on creation and compare against it.

Reply:
I was trying to preserve the order of SSRCs from the SDP to Jingle. If I sort `sources` in place, the newly added lines for `name` and `videoType` will be out of order.

Reply:
Aha. In that case it might be better to have a `const sortedSources = sources.slice().sort();` to avoid a copy on each loop iteration, which would be a bunch of them in case of many participants, no?